File: 4-Microservices/JkeWeb/src/main/resources/static/com/jke/widgets/InterestRate.js
/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2011. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:  
 * Use, duplication or disclosure restricted by GSA ADP Schedule 
 * Contract with IBM Corp. 
 *******************************************************************************/
dojo.provide("com.jke.widgets.InterestRate");

dojo.require("dijit._Widget");
dojo.require("dijit._Templated");
dojo.require("dijit.form.Select");
dojo.require("dijit.form.Button");
dojo.require("dojo.data.ItemFileReadStore");
dojo.require("dojo.currency");

dojo.declare("com.jke.widgets.InterestRate", [ dijit._Widget, dijit._Templated ], {

	// Path to the template
	templateString : dojo.cache("com.jke.widgets", "templates/InterestRate.html"),
	widgetsInTemplate : true,
	constructor : function(args) {
	},

	postCreate : function() {
        var i = 0;
        dojo.forEach(this.params.transactions, dojo.hitch(this, function(interestRate) {
            var tr, td;
            tr = dojo.create("tr", null, this.tableBody);
            if (i++ % 2 == 0)
                dojo.attr(tr, "style", "background-color: #efefef");

            td = dojo.create("td", null, tr);
            td.innerHTML = interestRate.category;

            td = dojo.create("td", null, tr);
            td.innerHTML = interestRate.tenure;

            td = dojo.create("td", null, tr);
            td.innerHTML = interestRate.interestRateRegular;

            td = dojo.create("td", null, tr);
            td.innerHTML = interestRate.interestRateSenior;
		}));

	}
});

File: 4-Microservices/JkeAPI/src/main/java/com/jke/jkeapi/jpa/InterestRate.java
/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2010, 2011. All Rights Reserved. 
 * 
 * Note to U.S. Government Users Restricted Rights:  Use, 
 * duplication or disclosure restricted by GSA ADP Schedule 
 * Contract with IBM Corp.
 *******************************************************************************/

package com.jke.jkeapi.jpa;


import javax.persistence.*;

@Table(name = "JKEINTERESTRATE")
@Entity
public class InterestRate {
    @Id
    @GeneratedValue
    private int id;
    private String category;
    private String tenure;
    @Column(name = "INTERESTRATEREGULAR")
    private String interestRateRegular;
    @Column(name = "INTERESTRATESENIOR")
    private String interestRateSenior;
    private String comments;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getTenure() {
        return tenure;
    }

    public void setTenure(String tenure) {
        this.tenure = tenure;
    }

    public String getInterestRateRegular() {
        return interestRateRegular;
    }

    public void setInterestRateRegular(String interestRateRegular) {
        this.interestRateRegular = interestRateRegular;
    }

    public String getInterestRateSenior() {
        return interestRateSenior;
    }

    public void setInterestRateSenior(String interestRateSenior) {
        this.interestRateSenior = interestRateSenior;
    }

    public String getComments() {
        return comments;
    }

    public void setComments(String comments) {
        this.comments = comments;
    }
}